pkg/ipamd/datastore/data_store.go

```diff
@@ -410,77 +454,99 @@ func (ds *DataStore) AddENI(eniID string, deviceNumber int, isPrimary, isTrunk,
 		return errors.New(DuplicatedENIError)
 	}
 	ds.eniPool[eniID] = &ENI{
-		createTime:    time.Now(),
-		IsPrimary:     isPrimary,
-		IsTrunk:       isTrunk,
-		IsEFA:         isEFA,
-		ID:            eniID,
-		DeviceNumber:  deviceNumber,
-		IPv4Addresses: make(map[string]*AddressInfo)}
+		createTime:         time.Now(),
+		IsPrimary:          isPrimary,
+		IsTrunk:            isTrunk,
+		IsEFA:              isEFA,
+		ID:                 eniID,
+		DeviceNumber:       deviceNumber,
+		AvailableIPv4Cidrs: make(map[string]*CidrInfo)}
+
 	enis.Set(float64(len(ds.eniPool)))
 	return nil
 }
 
-// AddIPv4AddressToStore add an IP of an ENI to data store
-func (ds *DataStore) AddIPv4AddressToStore(eniID string, ipv4 string) error {
+// AddIPv4AddressToStore add CIDR of an ENI to data store
+func (ds *DataStore) AddIPv4CidrToStore(eniID string, ipv4Cidr net.IPNet, isPrefix bool) error {
 	ds.lock.Lock()
 	defer ds.lock.Unlock()
 
+	strIPv4Cidr := ipv4Cidr.String()
+	ds.log.Infof("Adding %s to DS for %s", strIPv4Cidr, eniID)
 	curENI, ok := ds.eniPool[eniID]
 	if !ok {
+		ds.log.Infof("unkown ENI")
 		return errors.New("add ENI's IP to datastore: unknown ENI")
 	}
-
 	// Already there
-	_, ok = curENI.IPv4Addresses[ipv4]
+	_, ok = curENI.AvailableIPv4Cidrs[strIPv4Cidr]
 	if ok {
+		ds.log.Infof("IP already in DS")
 		return errors.New(IPAlreadyInStoreError)
 	}
 
-	ds.total++
-	// Prometheus gauge
+	curENI.AvailableIPv4Cidrs[strIPv4Cidr] = &CidrInfo{
```

Comment:
nit: it's seems ideal to me to have a variable captures `&CidrInfo ...` first. so we can avoid the map lookup in`ds.total += curENI.AvailableIPv4Cidrs[strIPv4Cidr].Size()`